scrape/scrape.go

```diff
@@ -1519,6 +1519,11 @@ loop:
 			// If any label limits is exceeded the scrape should fail.
 			if err = verifyLabelLimits(lset, sl.labelLimits); err != nil {
 				targetScrapePoolExceededLabelLimits.Inc()
+				level.Debug(sl.l).Log(
+					"msg", "Exceeded label limits",
+					"series", string(met),
```

Comment:
We should somehow limit this, if we hit the limit, it can be pretty long.

Reply:
> We should somehow limit this, if we hit the limit, it can be pretty long.

Use `github.com/prometheus/prometheus/util/logging.Deduper` in `scapePool`.
1. `Deduper` has an overhead that is a go-routine for cleaning, so put `Deduper` in `pool` instead of `loop`
2. `Deduper` encoding method can distinguish `keyvals` in Log. It does not omit logs that should output,  unless the `keyvals` on different lines are the same.